src/js/jsx/sections/layers/LayerFace.jsx

```diff
@@ -154,7 +144,15 @@ define(function (require, exports, module) {
          * @param {string} newName 
          */
         _handleLayerNameChange: function (event, newName) {
-            this.getFlux().actions.layers.rename(this.props.document, this.props.layer, newName);
+            this.getFlux().actions.layers.rename(this._getDocument(), this.state.layer, newName);
+        },
+        
+        /**
+         * @private
+         * @return {Document}
+         */
+        _getDocument: function () {
+            return this.getFlux().stores.document.getDocument(this.props.documentID);
```

Comment:
LayerFace uses this helper function to fetch the latest document whenever needed, instead of reading from its props. (actually the props do not include the document and layer models any more, they are replaced by their IDs)